controllers/controllers/services/instances/managed/controller.go

```diff
@@ -129,6 +117,18 @@ func (r *Reconciler) ReconcileResource(ctx context.Context, serviceInstance *kor
 		return ctrl.Result{}, fmt.Errorf("failed to create client for broker %q: %w", serviceBroker.Name, err)
 	}
 
+	if !serviceInstance.GetDeletionTimestamp().IsZero() {
+		return r.finalizeCFServiceInstance(ctx, osbapiClient, serviceInstance, serviceOffering, servicePlan)
```

Comment:
I think that we still need to move the finalization bit before getting getting plan/offering/etc

Our CI clusters could not be deleted as there are a couple of CFServiceInstances with finalizers that cannot be deleted. They have their deletion timestamp set but finalization fails with

```
    message: 'failed to get service plan "1ac1edee-804e-51bd-a4c0-149e8271f275": CFServicePlan.korifi.cloudfoundry.org "1ac1edee-804e-51bd-a4c0-149e8271f275" not found'
```

Indeed, there are no service plans on the cluster.

I do not know how the cluster reached this situation, but in any case, fainalizing resources should be "best effort" - they should try to delete whatever possible, but if an error occurs, still allow deletion. Having said that, I think we should move finalization at the top of the `Reconcile` method, just after setting the observed generation and have plan and offering being looked up in the finalize method. If an error occurrs while fetching the plan/offering occurs, it should be only logged AND the finalizer should be removed